server/src/test/java/org/cloudfoundry/identity/uaa/provider/oauth/ExternalOAuthAuthenticationManagerTest.java

```diff
@@ -154,16 +156,19 @@ public void cleanup() {
     }
 
     @Test
-    public void getExternalAuthenticationDetails_whenProviderHasSigningKey_throwsWhenIdTokenCannotBeValidated() {
+    public void getExternalAuthenticationDetails_whenProviderHasSigningKey_throwsWhenIdTokenCannotBeValidated() throws ParseException, JOSEException {
         expectedException.expect(InvalidTokenException.class);
         expectedException.expectMessage("Could not verify token signature.");
 
         Map<String, Object> header = map(
-                entry(HeaderParameterNames.ALGORITHM, JWSAlgorithm.HS256.getName()),
+                entry(HeaderParameterNames.ALGORITHM, JWSAlgorithm.RS256.getName()),
                 entry(HeaderParameterNames.KEY_ID, OIDC_PROVIDER_KEY)
         );
-        Signer signer = new RsaSigner(changedOidcProviderTokenSigningKey);
+        JWSSigner signer = new KeyInfo(OIDC_PROVIDER_KEY, oidcProviderTokenSigningKey, "http://localhost:8080/uaa").getSigner();
```

Comment:
The hard coded port 8080 looks concerning and should be centralized. we have run into issues were non default ports don't behave as expected or parameters to use different ports are ignored. This is a test and likely under more control, but good practice to keep them centralized and coordinated with startup/config